src/cryptoadvance/specter/wallet.py

```diff
@@ -529,36 +531,70 @@ def fetch_transactions(self):
+            if "alias" in json.loads(
+                address_labels
+            ):  # Key that is only present in Specter JSON
+                logger.debug("In the Specter JSON part.")
+                raw_dictionary = json.loads(address_labels)
+                labeled_addresses = {}
+                for label, address in raw_dictionary["labels"].items():
+                    labeled_addresses[address[0]] = label
+                logger.info(f"Specter JSON was converted to {labeled_addresses}.")
+            # Electrum
+            else:
+                logger.debug("In the Electrum JSON part.")
+                labeled_addresses = json.loads(address_labels)
+                logger.info(f"Electrum JSON was converted to {labeled_addresses}.")
+                # write tx_label to address_label in labels
+                for txitem in self._transactions.values():
+                    if txitem["txid"] not in labeled_addresses:
+                        continue
+                    address_list = (
+                        [txitem["address"]]
+                        if isinstance(txitem["address"], str)
+                        else txitem["address"]
+                    )
+                    for one_address in address_list:
+                        if one_address in labeled_addresses:
+                            continue  # if there is an address label, it supercedes the tx label
+                        labeled_addresses[one_address] = labeled_addresses[
+                            txitem["txid"]
+                        ]
+        # Specter CSV
+        except ValueError:  # If json.loads is not possible it throws a ValueError
+            labeled_addresses = {}
+            logger.debug(address_labels)
+            try:
+                f = StringIO(
+                    address_labels
+                )  # Drag & drop / pasting of CSV results in one giant string
+                dialect = csv.Sniffer().sniff(
+                    address_labels
+                )  # Delimiter is not always the same
+                reader = csv.DictReader(f, delimiter=dialect.delimiter)
+                reader.fieldnames = [name.lower() for name in reader.fieldnames]
+                for row in reader:
+                    labeled_addresses[row["address"]] = row["label"]
+                logger.info(f"Specter label CSV was converted to {labeled_addresses}.")
+            except (Error, KeyError) as e:
+                raise SpecterError(
+                    f"Labels import failed. Check the import info box for the expected formats. Error: {e}"
+                )
+        # convert labeled_addresses to arr (for AddressList.set_labels), only allow existing addresses for which a label had been set
         arr = [
             {"address": address, "label": label}
             for address, label in labeled_addresses.items()
-            if address in self._addresses
+            if address in self._addresses and self._addresses[address].is_labeled
```

Comment:
While this change might be the intended behavior for your specter-use-case, it is NOT for the electrum use case.  This could be copied and moved to the if statements to facilitate different behavior.